vehicle/src/Vehicle/Backend/LossFunction/Compile.hs

```diff
@@ -108,6 +113,18 @@ type MonadCompileLoss m =
     MonadReader (V.NetworkContext, DifferentiableLogic, V.DeclProvenance) m
   )
 
+flattenAnds :: InputArg -> NonEmpty InputArg
+flattenAnds arg = case argExpr arg of
+  V.AndExpr _ [e1, e2] -> flattenAnds e1 <> flattenAnds e2
+  _ -> [arg]
```

Comment:
It might be tidier to make `flattenAnds` and `flattenOrs` sub-functions of `reformatLogicalOperators` in the same way that `lowerNot` is.

Reply:
Will do.